### PR TITLE
custom table buttons

### LINKED ISSUE #95: Update uihints for adding buttons to tables in properties
schema changes required for https://github.com/elyra-ai/canvas/issues/901

### PR DESCRIPTION
**Pull request by caritaou**
closes #95 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: common-pipeline/operators/uihints-v3-schema.json [hunk 1]
```diff
@@ -652,6 +659,54 @@
...
+          "type": "string"
+        },
+        "tooltip": {
+          "description": "Tooltip text to display when the button is hovered",
+          "type": "object",
+          "$ref": "#/definitions/resource_definition"
+        }
+      },
+      "anyOf": [
+        {
```
Comment (matthoward366) [inline common-pipeline/operators/uihints-v3-schema.json:691]: Changes look good.  Once question is here.  Wouldn't either `icon or carbonIcon` always be required?

Comment (caritaou) [reply common-pipeline/operators/uihints-v3-schema.json:691]: If no icon is specified, then the button will just be a label. There are instances where the app does not want any icons

File: common-pipeline/operators/uihints-v3-schema.json [hunk 2]
```diff
@@ -662,6 +669,65 @@
...
+            "id",
+            "icon",
+            "description"
+          ]
+        },
+        {
+          "required": [
+            "id",
+            "carbon_icon",
+            "description"
```
Comment (caritaou) [inline common-pipeline/operators/uihints-v3-schema.json:727]: Carbon icons require a description to be displayed in the tooltip. Making this required for any icon only button to stay consistent

File: common-pipeline/operators/uihints-v3-schema.json [hunk 3]
```diff
@@ -662,6 +669,70 @@
...
+        },
+        "icon": {
+          "description": "URL to .svg image to display",
+          "type": "string"
+        },
+        "carbon_icon": {
+          "description": "Host provided name of Carbon icon to display. A callback function is required for the host application to return the jsx icon object imported from @carbon/icons-react library",
+          "type": "object"
+        },
+        "enable": {
```
Comment (matthoward366) [inline common-pipeline/operators/uihints-v3-schema.json:699]: Should this be enabled instead?

Comment (caritaou) [reply common-pipeline/operators/uihints-v3-schema.json:699]: "enabled" does sound better, but I was trying to keep consistent with what the toolbar config object expects https://github.com/elyra-ai/canvas/wiki/2.1-Config-Objects#toolbar-config-object

Should I go ahead and change it to "enabled"?

Comment (matthoward366) [reply common-pipeline/operators/uihints-v3-schema.json:699]: I think so since users don't really know about the toolbar-config in common-properties.

**Review by matthoward366 — APPROVED** [review]

**Pull request merged**